docs/branches.md

```diff
@@ -41,6 +40,68 @@ branch `main`. If the local target had also moved compared to `main@origin`
 merged. If one is ahead of the other, then that target will be the new target.
 Otherwise, the local branch will be conflicted (see next section for details).
 
+As of December 2023 Jujutsu tracks[^1] and fetches all branches by default, 
+which is confusing users coming from Git. To smoothen the transition branch 
+tracking was introduced. 
+
+### What does `git.auto-local-branch` actually do? 
+
+Jujutsu fetches all Git Refs under `/refs/remotes/origin` or if you have 
+multiple remotes `/refs/remotes/<remote name>` for each remote by default and 
+then stores them as remote tracking branches and creates local branches 
```

Comment:
I find this sentence confusing compared to my suggestion. In particular,

- It'd be nice to specify that we're talking about `jj git fetch` specifically.
- There are too many `and`s.
 
As an alternative to my previous suggestion you could say "Jujutsu's fetch operations consist of several steps. First, `jj git fetch` fetches all git refs under ... (or, if you are fetching from multiple remotes, ... for each remote). Then, Jujutsu stores these refs as remote-tracking branches. Finally, by default, Jujutsu creates local branches....

I'll leave this as non-blocking, since we'll be editing this later.

Reply:
Rewritten to your suggestion, I'm going to land this now and await your edits.